zerver/views/messages.py

```diff
@@ -1276,7 +1276,8 @@ def send_message_backend(request: HttpRequest, user_profile: UserProfile,
                          message_to: Union[Sequence[int], Sequence[str]]=REQ(
                              'to', type=Union[List[int], List[str]],
                              converter=extract_recipients, default=[]),
-                         forged: bool=REQ(default=False,
+                         forged: bool=REQ(validator=check_bool,
+                                          default=False,
```

Comment:
Noteworthy runtime behavior change: `?forged=false` would previously have been interpreted as truthy.

Reply:
Looking at the only code that uses this feature, I think we'll want to update that and/or do something hacky here to allow "yes" as a value :(.

```
tabbott@coset:~/python-zulip-api$ git grep forged
zulip/integrations/jabber/jabber_mirror_backend.py:            forged = "yes",
zulip/integrations/zephyr/zephyr_mirror_backend.py:        message["forged"] = "yes"
```

Reply:
Well, this will work as a transitional solution so that we can take some time in changing the mirroring code:

```
diff --git a/zerver/tests/test_messages.py b/zerver/tests/test_messages.py
index bd8354a4d6..7fe60f3cb6 100644
--- a/zerver/tests/test_messages.py
+++ b/zerver/tests/test_messages.py
@@ -2047,6 +2047,23 @@ class MessagePOSTTest(ZulipTestCase):
         msg = self.get_last_message()
         self.assertEqual(int(datetime_to_timestamp(msg.date_sent)), int(fake_timestamp))
 
+        # Now test again using forged=yes
+        fake_date_sent = timezone_now() - datetime.timedelta(minutes=22)
+        fake_timestamp = datetime_to_timestamp(fake_date_sent)
+
+        result = self.api_post(email, "/api/v1/messages", {"type": "stream",
+                                                           "forged": "yes",
+                                                           "time": fake_timestamp,
+                                                           "sender": "irc-user@irc.zulip.com",
+                                                           "content": "Test message",
+                                                           "client": "irc_mirror",
+                                                           "topic": "from irc",
+                                                           "to": "IRCLand"})
+        self.assert_json_success(result)
+
+        msg = self.get_last_message()
+        self.assertEqual(int(datetime_to_timestamp(msg.date_sent)), int(fake_timestamp))
+
     def test_unsubscribed_api_super_user(self) -> None:
         cordelia = self.example_user('cordelia')
         stream_name = 'private_stream'
diff --git a/zerver/views/messages.py b/zerver/views/messages.py
index 5b19e7b285..2b29676158 100644
--- a/zerver/views/messages.py
+++ b/zerver/views/messages.py
@@ -1276,9 +1276,9 @@ def send_message_backend(request: HttpRequest, user_profile: UserProfile,
                          message_to: Union[Sequence[int], Sequence[str]]=REQ(
                              'to', type=Union[List[int], List[str]],
                              converter=extract_recipients, default=[]),
-                         forged: bool=REQ(validator=check_bool,
-                                          default=False,
-                                          documentation_pending=True),
+                         forged_str: Optional[str]=REQ("forged",
+                                                       default=False,
+                                                       documentation_pending=True),
                          topic_name: Optional[str]=REQ_topic(),
                          message_content: str=REQ('content'),
                          widget_content: Optional[str]=REQ(default=None,
@@ -1296,6 +1296,10 @@ def send_message_backend(request: HttpRequest, user_profile: UserProfile,
                          tz_guess: Optional[str]=REQ('tz_guess', default=None,
                                                      documentation_pending=True)
                          ) -> HttpResponse:
+    # Temporary hack: We're transitioning `forged` from accepting
+    # `yes` to accepting `true` like all of our normal booleans.
+    forged = forged_str is not None and forged_str in ["yes", "true"]
+
     client = request.client
     is_super_user = request.user.is_api_super_user
     if forged and not is_super_user:
```